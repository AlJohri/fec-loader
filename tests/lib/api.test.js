const { ObjectReadableMock, ObjectWritableMock } = require('stream-mock');

jest.mock('cross-fetch');
const fetch = require('cross-fetch');

const api = require('../../lib/api');

test('api', async done => {
    const input = JSON.stringify({
        results: [
            {
                fec_file_id: 'FEC-24'
            }
        ]
    });

    const reader = new ObjectReadableMock(input);
    const writer = new ObjectWritableMock();

    fetch.mockReturnValue(
        Promise.resolve({
            body: reader
        })
    );

    const stream = await api({
        key: 'example'
    });

    stream.pipe(writer);

    writer.on('finish', () => {
        expect(fetch.mock.calls).toMatchSnapshot();
        expect(writer.data).toMatchSnapshot();
        done();
    });
    
});
