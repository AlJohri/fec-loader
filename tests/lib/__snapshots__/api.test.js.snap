// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api 1`] = `
Array [
  Array [
    "https://api.open.fec.gov/v1/efile/filings?sort=-receipt_date&per_page=100&page=1&api_key=example",
  ],
]
`;

exports[`api 2`] = `
Array [
  Object {
    "fec_file_id": 24,
  },
]
`;
